WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/RegisterDomainCoordinator.swift

```diff
@@ -29,75 +29,114 @@ class RegisterDomainCoordinator {
     func handleNoSiteChoice(on viewController: UIViewController) {
         createCart(
             onSuccess: { [weak self] in
-                self?.presentWebViewForNoSite(on: viewController)
+                self?.presentCheckoutWebview(on: viewController, title: TextContent.checkoutTitle, shouldPush: true)
             }) {
                 viewController.displayActionableNotice(title: TextContent.errorTitle, actionTitle: TextContent.errorDismiss)
             }
     }
 
+    /// Related to the `purchaseFromDomainManagement` Domain selection type.
+    /// Adds the selected domain to the cart then presents a site picker view.
     func handleExistingSiteChoice(on viewController: UIViewController) {
-        print("handleExistingSiteChoice")
+        let config = BlogListConfiguration(shouldShowCancelButton: false,
+                                           shouldShowNavBarButtons: false,
+                                           navigationTitle: TextContent.sitePickerNavigationTitle,
+                                           backButtonTitle: TextContent.sitePickerNavigationTitle,
+                                           shouldHideSelfHostedSites: true)
+        guard let blogListViewController = BlogListViewController(configuration: config, meScenePresenter: nil) else {
+            return
+        }
+
+        blogListViewController.blogSelected = { [weak self] controller, selectedBlog in
+            guard let self,
+                  let blog = selectedBlog else {
+                return
+            }
+            self.site = blog
+            controller?.showLoading()
+            self.createCart {
+                if let controller,
+                   let domainName = self.domain?.domainName {
+                    self.domainAddedToCartCallback?(controller, domainName, blog)
+                }
+                controller?.hideLoading()
+            } onFailure: {
+                controller?.displayActionableNotice(title: TextContent.errorTitle, actionTitle: TextContent.errorDismiss)
+                controller?.hideLoading()
+            }
+
+        }
+
+        viewController.navigationController?.pushViewController(blogListViewController, animated: true)
     }
 
     // MARK: Helpers
 
+    private func createCart(onSuccess: @escaping () -> (),
+                    onFailure: @escaping () -> ()) {
+        guard let domain else { return }
+        let siteID = site?.dotComID?.intValue
+        let proxy = RegisterDomainDetailsServiceProxy()
+        proxy.createPersistentDomainShoppingCart(siteID: siteID,
+                                                 domainSuggestion: domain.remoteSuggestion(),
+                                                 privacyProtectionEnabled: domain.supportsPrivacy ?? false,
+                                                 success: { _ in
+            onSuccess()
+        },
+                                                 failure: { _ in
+            onFailure()
```

Comment:
[Nit] If these can be passed directly as inline arguments rather than closures it might improve the readability. But I'm uncertain it is possible since it seem to be expecting an argument.

Reply:
> I'm uncertain it is possible since it seem to be expecting an argument.

Unfortunately, it is yes.